containers-tests/tests/intmap-properties.hs

```diff
@@ -1187,3 +1525,20 @@ prop_fromSet :: [(Int, Int)] -> Bool
 prop_fromSet ys =
   let xs = List.nubBy ((==) `on` fst) ys
   in fromSet (\k -> fromJust $ List.lookup k xs) (IntSet.fromList $ List.map fst xs) == fromList xs
+
+newtype Identity a = Identity a
+
+instance Functor Identity where
+  fmap f (Identity a) = Identity (f a)
+
+instance Applicative Identity where
+  pure a = Identity a
+  Identity f <*> Identity a = Identity (f a)
+
+prop_traverseWithKey :: IntMap () -> Bool
+prop_traverseWithKey mp = mp == newMap
+  where Identity newMap = traverseWithKey (\_ -> Identity) mp
```

Comment:
```haskell
-- Using Test.QuickCheck.Poly.A and .B and Test.QuickCheck.Function
prop_traverseWithKey :: Fun (Int, A) B -> IntMap A -> Property
prop_traverseWithKey fun mp = mp === newMap
  where Identity newMap = traverseWithKey (\k v -> Identity $ applyFun2 fun k v) mp
```

Reply:
This doesn't typecheck, since `(mp :: IntMap A) === (newMap :: IntMap B)`

Reply:
And with `Fun (Int, A) A` it fails with:
```
traverseWithKey2: [Failed]
*** Failed! Falsifiable (after 3 tests and 16 shrinks):
{_->1}
fromList [(-1,2)]
fromList [(-1,2)] /= fromList [(-1,1)]
```

Reply:
Because I wrote it wrong. What I meant was

```haskell
prop_traverseWithKey :: Fun (Int, A) B -> IntMap A -> Property
prop_traverseWithKey fun mp = mapWithKey (applyFun2 fun) mp === newMap
  where Identity newMap = traverseWithKey (\k v -> Identity $ applyFun2 fun k v) mp
```

Reply:
Ah, it was related to the other comment, gotcha. Already added this in the latest commit.